src/Libraries/Microsoft.Extensions.Diagnostics.Testing/Logging/FakeLogCollectorOptions.cs

```diff
@@ -34,6 +35,17 @@ public class FakeLogCollectorOptions
     /// </remarks>
     public ISet<LogLevel> FilteredLevels { get; set; } = new HashSet<LogLevel>();
 
+    /// <summary>
+    /// Gets or sets custom filters used to filter out records to be collected.
+    /// </summary>
+    /// <value>Default is an empty array.</value>
+    /// <remarks>
+    /// Defaults to an empty array, which doesn't filter any records.
+    /// If not empty, only records for which the filter function returns <see langword="true" /> will be collected by the fake logger.
```

Comment:
I think this is the point of the disconnect for me: "_only records for which the filter function returns <see langword="true" /> will be collected by the fake logger._". 
To me if a filter function returns true, it means the record was filtered out.

I recommend inverting the meaning here, which will invert the logic in the collector implementation too.